### ISSUE TITLE
rtmp推流，rtmp拉流同一条流，非集群模式， 如果推流断了，rtmp拉流是否要自动发起server close？

### ISSUE THREAD

**Issue by jinleileiking** [ossrs/srs#3000]
我看目前的逻辑是等待客户端断开rtmp拉流链接。 httpflv的逻辑是srs server主动断开，这块是否可以做成一致？

**Comment by najsword** [comment]
> 

你好，rtmp2rtc视频播放花屏有遇到吗，可以交流下吗

**Comment by najsword** [comment]
> 我看目前的逻辑是等待客户端断开rtmp拉流链接。 httpflv的逻辑是srs server主动断开，这块是否可以做成一致？
> 
> <img alt="image" width="970" src="https://user-images.githubusercontent.com/93968/162662311-656d6178-e79f-452d-b95c-4d8fc006a8c2.png">
> 
> 在flv拉流里有这个，但rtmp拉流没有

#3021

**Comment by winlinvip** [comment]
This is a problem that should be consistent.

If there is no disconnection, the client will not be aware when the anchor starts streaming again, which provides a better user experience.

However, FLV is difficult to maintain because it requires deleting the HTTP mount.

Of course, sometimes RTMP also needs to be disconnected in order to save resources.

So I think the better solution is: when RTMP configuration is disconnected for streaming, whether to also disconnect the playback, the default is to not disconnect and keep it consistent with the current situation.

I have marked this issue as `good first issue`, and everyone is welcome to provide a patch.

`TRANS_BY_GPT3`

**Comment by tangrufeng** [comment]
是的，我也期望能有推流端断开，拉流端http还能处于正常连接状态，而不是EOF

**Comment by tangrufeng** [comment]
> This is a problem, it should be consistent.
> 
> If there is no disconnection, the client will not be aware when the anchor starts streaming again, which provides a better user experience.
> 
> However, FLV is difficult to maintain because it requires deleting the HTTP mount.
> 
> Of course, sometimes RTMP also needs to be disconnected in order to save resources.
> 
> So I think the better solution is: when RTMP configuration is disconnected for streaming, whether to also disconnect for playback, the default is to not disconnect and keep it consistent with the current setting.
> 
> I have marked this issue as `good first issue`, and everyone is welcome to provide patches.

Yes, I also expect that the streaming end can be disconnected while the HTTP end remains in a normal connected state, instead of EOF.


`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
It's not possible to maintain consistent behavior across all protocols, so I suppose the inconsistency is acceptable.
